src/Profiles/Shifting/Shifter/ANTPLUS_ProfileShiftingShifter.cpp

```diff
@@ -0,0 +1,50 @@
+#include <Profiles/Shifting/Shifter/ANTPLUS_ProfileShiftingShifter.h>
+#include <CommonDataPages/ANTPLUS_CommonDataPagePrivateDefines.h>
+
+
+ProfileShiftingShifter::ProfileShiftingShifter(	uint16_t deviceNumber, uint8_t transmissionType) :
+    BaseMasterProfile(deviceNumber, transmissionType),
+    _patternStep( 0 ),
+    _toggle(0),
+    _eventCount( 0 )
+{
+    setChannelConfig();
+}
+
+void ProfileShiftingShifter::setChannelConfig() {
+    setChannelType(ANTPLUS_SHIFTING_MASTER_CHANNELTYPE);
+    setDeviceType(ANTPLUS_SHIFTING_DEVICETYPE);
+    setChannelPeriod(ANTPLUS_SHIFTING_CHANNELPERIOD);
+    setSearchTimeout(ANTPLUS_SHIFTING_SEARCHTIMEOUT);
+}
+
+void ProfileShiftingShifter::transmitNextDataPage() {
+
+    // some static aux messages (TODO preliminary solution)
+    const uint8_t manufacturer[]  = { 0x50, 0xFF, 0xFF, 0x01, 0x0F, 0x00, 0x85, 0x83 };
```

Comment:
What we will need to do here is setup a hook for the common datapages as well. Looks like I haven't done these messages yet. 😛

Reply:
To be honest: I did not check, whether there is some library code that generates these messages. A warapper class should be trivial.

Reply:
TODO added. No further changes planned by me for the next time.